File: 백준_2023_1월/backjoon/backjoon/stack10828.cpp
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <string.h>
#include <stack>

using namespace std;

stack<int> st;

int main(void) {

	int n;
	char comd[10] = {0};

	scanf("%d", &n);

	int arg;
	for (int i = 0; i < n; i++) {
		scanf("%s", comd);//명령 입력받기

		if (strcmp(comd, "push") == 0) {
			scanf("%d", &arg);
			st.push(arg);
		}
		else if (strcmp(comd, "pop") == 0) {
			if (st.empty()) {
				printf("-1\n");
				continue;
			}
			printf("%d\n", st.top());
			st.pop();
		}
		else if (strcmp(comd, "size") == 0) {
			printf("%d\n", st.size());
		}
		else if (strcmp(comd, "empty") == 0) {
			printf("%d\n", st.empty());
		}
		else {
			if (st.empty()) {
				printf("-1\n");
				continue;
			}
			printf("%d\n", st.top());
		}
	}
}

File: 코테_2023_1월/src/Stack10828.java
import java.util.Scanner;
import java.util.Stack;

public class Stack10828 {

    public static void main(String[] args){

        int n, arg;
        String comd;
        Stack<Integer> st = new Stack<>();

        Scanner sc = new Scanner(System.in);

        n = sc.nextInt();

        for(int i=0; i<n; i++){
            comd = sc.nextLine();

            if(comd.contains("push")){
                String[] strs = comd.split(" ");
                st.push(Integer.parseInt(strs[1]));
            }
            else if(comd.equals("pop")){
                if(st.empty()){
                    System.out.println("-1");
                }
                else{
                    int temp;
                    temp = st.pop();
                    System.out.println(temp);
                }
            }
            else if(comd.equals("size")){
                System.out.println(st.size());
            }
            else if(comd.equals("empty")){
                System.out.println(st.empty() ? "1" : "0");
            }
            else if(comd.equals("top")){
                if(st.empty()){
                    System.out.println("-1");
                }
                else {
                    System.out.println(st.peek());
                }
            }
        }

    }
}
